modules/op-about-whenexpression.adoc

```diff
@@ -0,0 +1,174 @@
+        taskSpec:
+          params:
+            - name: path
+          workspaces:
+            - name: source
+              description: The workspace to check for the file
+          results:
+            - name: exists
+              description: indicates whether the file exists or is missing
+          steps:
+            - name: check-file
+              image: alpine
+              script: |
+                if test -f $(workspaces.source.path)/$(params.path); then
+                  printf yes | tee /tekton/results/exists
+                else
+                  printf no | tee /tekton/results/exists
+                fi
+      - name: echo-file-exists
+        when: <3>
+          - input: "$(tasks.check-file.results.exists)"
+            operator: in
+            values: ["yes"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: 'echo file exists'
+...
+      - name: task-should-be-skipped-1
+        when: <4>
+          - input: "$(params.path)"
+            operator: notin
+            values: ["README.md"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: exit 1
+...
+    finally:
+      - name: finally-task-should-be-executed
+        When: <5>
+          - input: "$(tasks.echo-file-exists.status)"
+            operator: in
+            values: ["Succeeded"]
+          - input: "$(tasks.status)"
+            operator: in
+            values: ["Succeeded"]
+          - input: "$(tasks.check-file.results.exists)"
+            operator: in
+            values: ["yes"]
+          - input: "$(params.path)"
+            operator: in
+            values: ["README.md"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: 'echo finally done'
+  params:
+    - name: path
+      value: README.md
+  workspaces:
+    - name: source
+      volumeClaimTemplate:
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 16Mi
+----
+<1> Specifies the type of Kubernetes object. In this example, `PipelineRun`.
+<2> Specifies the list of Tasks used in the Pipeline.
+<3> Specifies the `when` expression to execute the `echo-file-exists` task only if the `exists` result from the `check-file` task is `yes`.
+<4> Specifies the `when` expression to skip the `task-should-be-skipped-1` task only if the `path` parameter is not `README.md`.
+<5> Specifies the `when` expression to execute the `finally-task-should-be-executed` task only if the execution status of the `echo-file-exists` task and the task status is `Succeeded`, the `exists` result from the `check-file` task is `yes`, and the `path` parameter is `README.md`.
+
+The following example shows the `when` expressions to execute the `run-lint` task only if the `lint-config` optional workspace has been provided by a PipelineRun.
```

Comment:
Does this example add any additional information other than the above example? If not, we can skip it.

Reply:
remove to. It shows the expression executing the task. 
Is provided. Try to keep all software actions in the present tense. (See page 5 of the style guide)